controllers/report_controller.go

```diff
@@ -107,3 +154,217 @@ func requeueReportsHandler(c client.Client, log logr.Logger) handler.EventHandle
 		return res
 	})
 }
+
+type reportSchedule interface {
```

Comment:
Can we move all the metering function's into their own file so we don't clutter the report controller file?